src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/command/rulemanager/OfSpeakerService.java

```diff
@@ -51,74 +57,84 @@ public OfSpeakerService(@NonNull FloodlightModuleContext moduleContext) {
-        for (OfCommand data : request.getCommands()) {
-            data.buildInstall(holder, switchId);
-        }
-        OfBatchExecutor executor = OfBatchExecutor.builder()
-                .iofSwitch(sw)
-                .kafkaUtilityService(kafkaUtilityService)
-                .kafkaProducerService(kafkaProducerService)
-                .sessionService(sessionService)
-                .messageContext(request.getMessageContext())
-                .holder(holder)
-                .switchFeatures(featureDetectorService.detectSwitch(sw))
-                .kafkaKey(key)
-                .origin(request.getOrigin())
-                .failIfExists(request.isFailIfExists())
-                .build();
-        executor.executeBatch();
+        processBatchRequest(request, key, this::buildInstallOfCommand, request.isFailIfExists());
     }
 
     @Override
     public void processBatchModify(ModifySpeakerCommandsRequest request, String key) {
-        SwitchId switchId = request.getSwitchId();
-        DatapathId dpId = DatapathId.of(switchId.toLong());
-        IOFSwitch sw = iofSwitchService.getSwitch(dpId);
-        OfBatchHolder holder = new OfBatchHolder(iofSwitchService, request.getMessageContext(),
-                request.getCommandId(), request.getSwitchId());
-        for (OfCommand data : request.getCommands()) {
-            data.buildModify(holder, switchId);
-        }
-        OfBatchExecutor executor = OfBatchExecutor.builder()
-                .iofSwitch(sw)
-                .kafkaUtilityService(kafkaUtilityService)
-                .kafkaProducerService(kafkaProducerService)
-                .sessionService(sessionService)
-                .messageContext(request.getMessageContext())
-                .holder(holder)
-                .switchFeatures(featureDetectorService.detectSwitch(sw))
-                .kafkaKey(key)
-                .origin(request.getOrigin())
-                .build();
-        executor.executeBatch();
+        processBatchRequest(request, key, this::buildModifyOfCommand, true);
     }
 
     @Override
     public void processBatchDelete(DeleteSpeakerCommandsRequest request, String key) {
+        processBatchRequest(request, key, this::buildDeleteOfCommand, true);
+    }
+
+    private void processBatchRequest(BaseSpeakerCommandsRequest request, String key,
+                                     OfCommandProcessor processor, boolean failIfExists) {
         SwitchId switchId = request.getSwitchId();
         DatapathId dpId = DatapathId.of(switchId.toLong());
         IOFSwitch sw = iofSwitchService.getSwitch(dpId);
         OfBatchHolder holder = new OfBatchHolder(iofSwitchService, request.getMessageContext(),
                 request.getCommandId(), request.getSwitchId());
         for (OfCommand data : request.getCommands()) {
-            data.buildDelete(holder, switchId);
+            processor.process(data, holder, switchId);
+        }
+        if (sw == null) {
```

Comment:
`processor.process(...)` will make `sw` lookup by itself, will report error into log and alter `processor` internal state. So this null check performed too late.

It should be reworked in some way, but I don't have an specific proposition for it right now.